doc/source/reference/array_api.rst

```diff
@@ -1,737 +1,64 @@
+documentation of the array API standard itself and to code and
+developer-focused documention in SciPy and scikit-learn.
 
-The ``numpy.array_api`` module serves as a minimal, reference implementation
-of the array API standard. In being minimal, the module only implements those
-things that are explicitly required by the specification. Certain things are
-allowed by the specification but are explicitly disallowed in
-``numpy.array_api``. This is so that the module can serve as a reference
-implementation for users of the array API standard. Any consumer of the array
-API can test their code against ``numpy.array_api`` and be sure that they
-aren't using any features that aren't guaranteed by the spec, and which may
-not be present in other conforming libraries.
+Note that in order to use standard-complaint code with older NumPy versions
+(< 2.0), the `array-api-compat
+<https://github.com/data-apis/array-api-compat>`__ package may be useful. For
+testing whether NumPy-using code is only using standard-compliant features
+rather than anything NumPy-specific, the `array-api-strict
+<https://github.com/data-apis/array-api-strict>`__ package can be used.
 
-The ``numpy.array_api`` module is not documented here. For a listing of the
-functions present in the array API specification, refer to the `array API
-standard <https://data-apis.org/array-api/latest/>`__. The ``numpy.array_api``
-implementation is functionally complete, so all functionality described in the
-standard is implemented.
+.. admonition:: History
 
-.. _array_api-differences:
+    NumPy 1.22.0 was the first version to include support for the array API
+    standard, via a separate ``numpy.array_api`` submodule. This module was
+    marked as experimental (it emitted a warning on import) and removed in
+    NumPy 2.0 because full support was included in the main namespace.
+    `NEP 47 <https://numpy.org/neps/nep-0047-array-api-standard.html>`__ and
+    `NEP 56 <https://numpy.org/neps/nep-0056-array-api-main-namespace.html>`__
+    describe the motivation and scope for implementing the array API standard
+    in NumPy.
 
-Table of differences between ``numpy.array_api`` and ``numpy``
-==============================================================
 
-This table outlines the primary differences between ``numpy.array_api`` from
-the main ``numpy`` namespace. There are three types of differences:
+Entry point
+===========
 
-1. **Strictness**. Things that are only done so that ``numpy.array_api`` is a
-   strict, minimal implementation. They aren't actually required by the spec,
-   and other conforming libraries may not follow them. In most cases, spec
-   does not specify or require any behavior outside of the given domain. The
-   main ``numpy`` namespace would not need to change in any way to be
-   spec-compatible for these.
+NumPy installs an `entry point <https://packaging.python.org/en/latest/specifications/entry-points/>`__
+that can be used for discovery purposes::
```

Comment:
I never noticed that this was actually added to the standard. I should probably add it to array-api-compat and array-api-strict.

Reply:
It's marked as optional and not in the API docs, since it isn't actually an API but a packaging thing. Maybe we should improve its visibility though?